src/main/kotlin/org/jitsi/nlj/RtpLayerDesc.kt

```diff
@@ -123,14 +147,38 @@ constructor(
     fun getBitrateBps(nowMs: Long): Long {
         var bitrate = rateStatistics.getRate(nowMs)
 
-        /* TODO: does the wrong thing if we have multiple dependencies */
+        val rates = HashMap<Int, Long>()
+
+        getBitrateBps(nowMs, rates)
+
+        return rates.values.sum()
+    }
+
+    /**
+     * Recursively adds the bitrate (in bps) of this [RTPLayerDesc] and
+     * its dependencies in the map passed in as an argument.
+     *
+     * This is necessary to ensure we don't double-count layers in cases
+     * of multiple dependencies.
+     *
+     * @param nowMs
+     */
+    private fun getBitrateBps(nowMs: Long, rates: MutableMap<Int, Long>) {
+        if (rates.containsKey(index)) {
+            return
+        }
+        rates[index] = rateStatistics.getRate(nowMs)
+
         if (dependencyLayers != null) {
             for (dependency in dependencyLayers) {
-                bitrate += dependency.getBitrateBps(nowMs)
+                dependency.getBitrateBps(nowMs, rates)
+            }
+        }
+        if (useSoftDependencies && softDependencyLayers != null) {
```

Comment:
The issue is that we determine what the soft dependencies _are_ by parsing the scalability structure, but we don't determine whether soft dependencies are _in use_ until we see whether the upper spatial-layer packet references the lower one.

I can make the array be empty rather than nullable, though.